site/public/js/server.js

```diff
@@ -178,6 +178,42 @@ function newDeleteCourseMaterialForm(id, file_name) {
     form.find('.form-body').scrollTop(0);
 }
 
+function newOverwriteCourseMaterialForm(clashing_names, is_link, is_edit_form) {
+    const form = $('#overwrite-course-material-form');
+    form.css('display', 'block');
+    if (clashing_names.length === 1) {
+        let to_replace = [[' All', '']];
+        if (is_link) {
+            to_replace.push(['file', 'link']);
+        }
+        form.html((i, html) => {
+            to_replace.forEach((elem) => {
+                html = html.replaceAll(elem[0], elem[1]);
+            });
+            return html;
+        });
+    }
+    else {
+        const singular = ['Material', 'file', 'name', 'one'];
+        form.html((i, html) => {
+            singular.forEach((elem) => {
+                html = html.replaceAll(RegExp(`( a){0,1}( ${elem})([ .:])`, 'g'), '$2s$3');
+            });
+            return html;
+        });
+    }
+    var clash_list = $('#existing-names');
+    clash_list.html("");
+    clashing_names.forEach((elem) => {
+        clash_list.append(`<li>${elem}</li>`);
+    });
```

Comment:
Having any sort of html element like this is a bit worrying. Could you make it dynamically add html elements and set text content.

Reply:
Please see if the change is sufficient or something else would be needed.